crane-cloud/backend
# added expiration of credits after 6 months

# Description

Users credits now expire after 6 months

## Type of change

Please delete options that are not relevant.



## Trello Ticket ID

https://trello.com/c/pTc0nnpQ/1139-add-functionality-for-expiry-of-promotional-credits-expire-after-6-months

## How Can This Be Tested?

On server.py line 109 change the **hours** of the scheduler to 1 from 24 (or appropriate time frame). This will make the scheduler run every hour instead of every 24 hours. 

Run the backend and assign the user credits.

In the transactions table, change the month and day of the expiration date of the assigned credits of the user to the current date of testing. This will make it so that the credits expiration date is the current day.

Wait 1 hour (or appropriate time frame)  and credits should expire on next scheduled run.

Note: During normal usage, the scheduler will run every 24 hours to look for expired credits. A users credits expire six months after a users latest credit assignment date.

## Checklist:
